controllers/openshift_controller.go

```diff
@@ -620,6 +620,17 @@ func (r *KataConfigOpenShiftReconciler) getNodeSelectorAsMap() map[string]string
 	return nodeSelector
 }
 
+func (r *KataConfigOpenShiftReconciler) getNodeSelectorAsMapNoKataOc() map[string]string {
```

Comment:
Not sure to understand the semantics of this function and thus the last sentence of the commit message...

Reply:
It's basically `getNodeSelectorAsMap()` without the "kata-oc" requirement. With this PR we're moving away from "combined" node selectors that contain both the "canonical" label requirements coming from KataConfig (mostly `kataConfigPoolSelector`) _and_ the "kata-oc" node role derived from them.  This is a temporary function that returns just the "canonical" part to bridge the gap and keep the controller compilable until a proper implementation is in place.

Don't worry, this will be removed soon (I have a subsequent PR pretty much ready that will do that, pending on merge of this one).